dbms/src/Functions/FunctionsRound.h

```diff
@@ -556,6 +562,163 @@ class FunctionRounding : public IFunction
 };
 
 
+class FunctionRoundDown : public IFunction {
+public:
+    static constexpr auto name = "roundDown";
+    static FunctionPtr create(const Context & context) { return std::make_shared<FunctionRoundDown>(context); }
+    FunctionRoundDown(const Context & context) : context(context) {}
+
+public:
+    String getName() const override
+    {
+        return name;
+    }
+
+    bool isVariadic() const override { return false; }
+    size_t getNumberOfArguments() const override { return 2; }
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {1}; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override {
+        const DataTypePtr & type_x = arguments[0];
+
+        if (!isNumber(type_x))
```

Comment:
I would recommend to be consistent with one-line if blocks. You should either put all such blocks in brackets or none of them (line 608).